qw5q_gold_qblox.py

```diff
@@ -157,85 +34,82 @@ def create(runcard_path=RUNCARD):
-        module_settings = settings[name]
-        modules[name] = cls(name=name, address=address, settings=module_settings)
-        return modules[name]
-
+    runcard = load_runcard(runcard_path)
     modules = {}
 
     cluster = Cluster(
         name="cluster",
-        address="192.168.0.20",
-        settings=instruments_settings["cluster"],
+        address=ADDRESS,
+        settings=Cluster_Settings(reference_clock_source=ReferenceClockSource.INTERNAL),
     )
 
-    qrm_rf_a = instantiate_module(
-        modules, ClusterQRM_RF, "qrm_rf_a", "192.168.0.20:16", instruments_settings
-    )  # qubits q0, q1, q5
-    qrm_rf_b = instantiate_module(
-        modules, ClusterQRM_RF, "qrm_rf_b", "192.168.0.20:18", instruments_settings
-    )  # qubits q2, q3, q4
-
-    qcm_rf0 = instantiate_module(
-        modules, ClusterQCM_RF, "qcm_rf0", "192.168.0.20:6", instruments_settings
-    )  # qubit q0
-    qcm_rf1 = instantiate_module(
-        modules, ClusterQCM_RF, "qcm_rf1", "192.168.0.20:8", instruments_settings
-    )  # qubits q1, q2
-    qcm_rf2 = instantiate_module(
-        modules, ClusterQCM_RF, "qcm_rf2", "192.168.0.20:10", instruments_settings
-    )  # qubits q3, q4
-
-    qcm_bb0 = instantiate_module(
-        modules, ClusterQCM_BB, "qcm_bb0", "192.168.0.20:2", instruments_settings
-    )  # qubit q0
-    qcm_bb1 = instantiate_module(
-        modules, ClusterQCM_BB, "qcm_bb1", "192.168.0.20:4", instruments_settings
-    )  # qubits q1, q2, q3, q4
-
     # DEBUG: debug folder = report folder
     # import os
     # folder = os.path.dirname(runcard) + "/debug/"
     # if not os.path.exists(folder):
     #     os.makedirs(folder)
     # for name in modules:
     #     modules[name]._debug_folder = folder
+    qcm_bb0 = ClusterQCM_BB("qcm_bb0", f"{ADDRESS}:2", cluster)
+    qcm_bb1 = ClusterQCM_BB("qcm_bb1", f"{ADDRESS}:4", cluster)
+    qcm_rf0 = ClusterQCM_RF("qcm_rf0", f"{ADDRESS}:6", cluster)
+    qcm_rf1 = ClusterQCM_RF("qcm_rf1", f"{ADDRESS}:8", cluster)
+    qcm_rf2 = ClusterQCM_RF("qcm_rf2", f"{ADDRESS}:10", cluster)
+    qrm_rf_a = ClusterQRM_RF("qrm_rf_a", f"{ADDRESS}:16", cluster)
+    qrm_rf_b = ClusterQRM_RF("qrm_rf_b", f"{ADDRESS}:18", cluster)
 
     controller = QbloxController("qblox_controller", cluster, modules)
+
     twpa_pump = SGS100A(name="twpa_pump", address="192.168.0.36")
+    instruments = {
+        controller.name: controller,
+        twpa_pump.name: twpa_pump,
+        qcm_bb0.name: qcm_bb0,
+        qcm_bb1.name: qcm_bb1,
+        qcm_rf0.name: qcm_rf0,
+        qcm_rf1.name: qcm_rf1,
+        qcm_rf2.name: qcm_rf2,
+        qrm_rf_a.name: qrm_rf_a,
+        qrm_rf_b.name: qrm_rf_b,
+    }
+
+    instruments = load_instrument_settings(runcard, instruments)
+
+    modules["qcm_bb0"] = qcm_bb0
+    modules["qcm_bb1"] = qcm_bb1
+    modules["qcm_rf0"] = qcm_rf0
+    modules["qcm_rf1"] = qcm_rf1
+    modules["qcm_rf2"] = qcm_rf2
+    modules["qrm_rf_a"] = qrm_rf_a
+    modules["qrm_rf_b"] = qrm_rf_b
```

Comment:
Couldn't you generate `modules` just by filtering `instruments`? Are they conceptually different from each other?

Reply:
Thanks @AleCandido for the tip, I was focused on other thins and I didn't thought about it. Of course you can, I will modify this asap.